src/database/postgres_test.go

```diff
@@ -121,90 +121,58 @@ func TestPostgres(t *testing.T) {
 
 	db, err := createTestingDatabase()
 	if err != nil {
-		t.Fatalf("Failed to create test database: %v", err)
+		fmt.Printf("Failed to create test database: %v", err)
+		os.Exit(1)
 	}
 	defer db.Close()
 
-	// Test connection
+	m.Run()
```

Comment:
Why change this to testing.M ? It seemed fine the way it was. It's better in general to test as testing.T whenever possible

Reply:
I have to initialise the test database once and then run all the tests. I used to do the initialisation and then remove the test db for every test case which was an overkill. So moved it to testing.M

Reply:
The problem with reusing the db is that it violates hermiticity (even if it does make the test faster). We can leave it like this for now I suppose though